endpoint.go

```diff
@@ -777,7 +766,7 @@ func (ep *endpoint) Delete(force bool) error {
 
 	ep.releaseAddress()
 
-	return nil
+	return n.getEpCnt().DecEndpointCnt()
```

Comment:
Which means we are inconsistent in the handling of ep.Delete errors in engine and IMO we should fix it. For example here https://github.com/docker/docker/blob/master/daemon/container_operations.go#L763 we just log a warning. It should be the caller who should decide whether this error should be logged are not. `Maskable` kind of logic should be used sparingly.